src/pages/workspace/rules/RulesCustomNamePage.tsx

```diff
@@ -86,23 +114,28 @@ function RulesCustomNamePage({route}: RulesCustomNamePageProps) {
                     style={[styles.flexGrow1, styles.mh5]}
                     formID={ONYXKEYS.FORMS.RULES_CUSTOM_NAME_MODAL_FORM}
                     validate={validateCustomName}
-                    onSubmit={({customName}) => {
-                        setPolicyDefaultReportTitle(policyID, customName);
-                        Navigation.setNavigationActionToMicrotaskQueue(Navigation.goBack);
-                    }}
+                    onSubmit={submitForm}
                     submitButtonText={translate('common.save')}
                     enabledWhenOffline
                     shouldHideFixErrorsAlert
+                    isLoading={isLoading}
                     addBottomSafeAreaPadding
                 >
-                    <InputWrapper
-                        InputComponent={TextInput}
-                        inputID={INPUT_IDS.CUSTOM_NAME}
-                        defaultValue={customNameDefaultValue}
-                        label={translate('workspace.rules.expenseReportRules.customNameInputLabel')}
-                        aria-label={translate('workspace.rules.expenseReportRules.customNameInputLabel')}
-                        ref={inputCallbackRef}
-                    />
+                    <OfflineWithFeedback
+                        pendingAction={policy?.fieldList?.[CONST.POLICY.FIELDS.FIELD_LIST_TITLE]?.pendingFields?.defaultValue}
+                        errors={reportTitleErrors}
```

Comment:
Shouldn't we pass the report title error specifically rather than any error related to the field list. For example, I think we could just pass the title error variable you have directly. Also, when clearing errors, we should clear from the most specific field path possible, so clearing errors for the text title field instead of the whole field list. Or is there a good reason you have done it this way? Is it because the fieldList is an array?

I think you might find this helpful because our error handling can be a little bit complicated.

https://github.com/Expensify/App/blob/59a07d6b8ba20b13cdaf5e3dc388a8bc8f33f61e/contributingGuides/API.md#L50-L66